packages/react/runtime/src/lynx/calledByNative.ts

```diff
@@ -13,8 +13,51 @@ import { renderMainThread } from '../lifecycle/render.js';
 import { hydrate } from '../hydrate.js';
 import { markTiming, PerformanceTimingKeys, setPipeline } from './performance.js';
 import { __pendingListUpdates } from '../list.js';
+import { ssrHydrateByOpcodes } from '../opcodes.js';
+
+function ssrEncode() {
+  const { __opcodes } = __root;
+  delete __root.__opcodes;
+
+  const oldToJSON = SnapshotInstance.prototype.toJSON;
+  SnapshotInstance.prototype.toJSON = function(this: SnapshotInstance): any {
+    return [
+      this.type,
+      this.__id,
+      this.__elements,
+    ];
+  };
+
+  try {
+    return JSON.stringify({ __opcodes, __root_values: __root.__values });
+  } finally {
+    SnapshotInstance.prototype.toJSON = oldToJSON;
+  }
+}
+
+function ssrHydrate(info: string) {
+  const nativePage = __GetPageElement();
+  if (!nativePage) {
+    throw 'SSR Hydration Failed! Please check if the SSR content loaded successfully!';
```

Comment:
I'm not convinced that throwing an error is the best approach. While we lack a call stack here (as `ssrHydrate` is invoked natively), there might still be scenarios where someone expects an error to be thrown.

Reply:
```suggestion
    throw new Error('SSR Hydration Failed! Please check if the SSR content loaded successfully!');
```